semantic_router/llms/llamacpp.py

```diff
@@ -0,0 +1,70 @@
+from contextlib import contextmanager
+from pathlib import Path
+from typing import Any
+
+from llama_cpp import Llama, LlamaGrammar
+
+from semantic_router.llms.base import BaseLLM
+from semantic_router.schema import Message
+from semantic_router.utils.logger import logger
+
+
+class LlamaCppLLM(BaseLLM):
+    llm: Llama | None
+    temperature: float | None
+    max_tokens: int | None
+    grammar: LlamaGrammar | None
+
+    def __init__(
+        self,
+        llm: Llama,
```

Comment:
why don't we initialize the LLM within the class for those that don't know how to use Llama.cpp (granted, it's not complex)

Reply:
@bruvduroiu tagging you here so you see it, but resolving for now